queue/abstract.lua

```diff
@@ -393,45 +388,46 @@ local function make_self(driver, space, tube_name, tube_type, tube_id, opts)
     return self
 end
 
+--- Release all session tasks.
+local function release_session_tasks(session_uuid)
+    local taken_tasks = box.space._queue_taken_2.index.uuid:select(session_uuid)
+
+    for _, task in pairs(taken_tasks) do
```

Comment:
Since `pairs `can't be JIT-compiled[1] it's preferable to use `ipairs` here

[1] http://wiki.luajit.org/NYI#libraries_base-library

Reply:
Thank you for the information. This code has been moved in "refactoring: move" session release tasks "to a separate function" and I don't see any reasons to change this logic within the current task. But if we will have benchmark that will demonstrate profit from such changes, they can be implemented by a separate PR.

Reply:
I think changing pairs to ipairs we could consider as refactoring. Small note in commit message is more than enough IMO.

Reply:
```
local clock = require('clock')

local t = {}

for i = 1, 1e8 do
    table.insert(t, i)
end

local start = clock.time()
for _ in pairs(t) do end
print('pairs ', clock.time() - start)

collectgarbage()
collectgarbage()

local start = clock.time()
for _ in ipairs(t) do end
print('ipairs ', clock.time() - start)

os.exit()
```

Results:
```
pairs 	0.36381793022156
ipairs 	0.071022033691406
```

Reply:
Let's test it with `jit.off()`
```
$ tarantool bench1.lua
pairs 	1.0151600837708
ipairs 	2.2498149871826
```

Reply:
Sorry, but this is syntetic bullshit that doesn't relate to the actual code. Lua code might fail to be compiled due to other NYIs within loop body or even particular data being processed (see tarantool/tarantool#4630). Do you have any benchmarks and trace dumps for queue (at least) or rather performance critical application using this module? If not, the let's stop the discussion right here.

By the way, Leonid arguments are more reasonable than unproven performance degradation.

Reply:
OK, let's stay it as is.